testutils/utils.go

```diff
@@ -79,3 +81,29 @@ func ToJsonOrDie(value interface{}) []byte {
 		panic(err)
 	}
 }
+
+func WaitForAlertManagerSendGridReceiver(
+	k8sClient client.Client,
+	ctx context.Context,
+	amconfigKey client.ObjectKey,
+	emailadresses []string,
+	timeout time.Duration,
+	interval time.Duration,
+) {
+	var resIndex int
+	EventuallyWithOffset(1, func() string {
+		amconfig := &promv1a1.AlertmanagerConfig{}
+		ExpectWithOffset(2, k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
+		for i := range amconfig.Spec.Receivers {
+			reciever := &amconfig.Spec.Receivers[i]
+			if reciever.Name == "SendGrid" {
+				resIndex = i
+				break
+			}
+		}
+		if len(amconfig.Spec.Receivers[resIndex].EmailConfigs) > 0 {
+			return amconfig.Spec.Receivers[resIndex].EmailConfigs[0].To
+		}
+		return ""
+	}, timeout, interval).Should(Equal(strings.Join(emailadresses, ", ")))
```

Comment:
You have a couple of issues with this util function:
1. You do not protect against a receiver list that does not contain the desired reciver.  
2. You find the receiver ref but then you just throw it and dave the index just to reindex into the Receivers array.


```suggestion
	EventuallyWithOffset(1, func() string {
		amconfig := &promv1a1.AlertmanagerConfig{}
		ExpectWithOffset(2, k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
		var reciever *RecieverType = nullptr;
		for i := range amconfig.Spec.Receivers {
			r := &amconfig.Spec.Receivers[i]
			if r.Name == "SendGrid" {
				reciever = r
				break
			}
		}

		if reciver != nullptr && len(amconfig.Spec.Receivers[resIndex].EmailConfigs) > 0 {
			return amconfig.Spec.Receivers[resIndex].EmailConfigs[0].To
		}
		return ""
	}, timeout, interval).Should(Equal(strings.Join(emailadresses, ", ")))
```

or even better and much simpler:
```suggestion
	EventuallyWithOffset(1, func() string {
		amconfig := &promv1a1.AlertmanagerConfig{}
		ExpectWithOffset(2, k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
		for i := range amconfig.Spec.Receivers {
			reciever := &amconfig.Spec.Receivers[i]
			if reciever.Name == "SendGrid" {
				if len(amconfig.Spec.Receivers[resIndex].EmailConfigs) > 0 {
					return amconfig.Spec.Receivers[resIndex].EmailConfigs[0].To
				} else {
				        return ""
				}
			}
		}
		return ""
	}, timeout, interval).Should(Equal(strings.Join(emailadresses, ", ")))
```